### PR TITLE
Remove mute/block/report from menu for own toots

### LINKED ISSUE #519: App lets user Block/Mute/Report themself
When looking at my own toot, the gear icon gives options to Block/Mute/Report/Delete. 

Delete makes sense. Block/Mute/Report doesn’t, and could lead to users accidentally blocking themselves.

### PR DESCRIPTION
**Pull request by nossipova**
This PR fixes #519.

I implemented the most simple solution. If something more generic is needed or if I overlooked something (it's my first contribution here), please let me know.

### CODE REVIEW

File: MastodonSDK/Sources/MastodonUI/View/Content/StatusAuthorView.swift [hunk 1]
```diff
@@ -152,25 +152,30 @@ extension StatusAuthorView {
...
+            ])
+        }
+        
+        actions.append(contentsOf: [
             .bookmarkStatus(
                 .init(isBookmarking: menuContext.isBookmarking)
             ),
             .shareStatus
-        ]
+        ])
```
Comment (Yonodactyl) [inline MastodonSDK/Sources/MastodonUI/View/Content/StatusAuthorView.swift:178]: Was your placement of this append to make sure the order of the `MastodonMenu.Action` array matches the procedural ordering?

**Comment by Yonodactyl** [comment]
https://github.com/mastodon/mastodon-ios/blob/38221e059972776b666e9eb84ccf2d1bb686329a/MastodonSDK/Sources/MastodonUI/View/Content/StatusAuthorView.swift#L172-L179

Curious about the process, no criticizing or reviewing: were you spacing your appends below and above this to make sure that the `MastodonMenu.Actions` array followed a procedural ordering?

**Comment by nossipova** [comment]
> Curious about the process, no criticizing or reviewing: were you spacing your appends below and above this to make sure that the `MastodonMenu.Actions` array followed a procedural ordering?

Exactly. I didn't want to change the order of the menu items without a reason.

File: MastodonSDK/Sources/MastodonUI/View/Content/StatusAuthorView.swift [hunk 2]
```diff
@@ -152,25 +152,30 @@ extension StatusAuthorView {
...
-                name: menuContext.name,
-                isBlocking: menuContext.isBlocking
-            )),
-            .reportUser(
-                .init(name: menuContext.name)
-            ),
+        var actions = [MastodonMenu.Action]()
+
+        if !menuContext.isMyself {
+            actions.append(contentsOf: [
```
Comment (kimar) [inline MastodonSDK/Sources/MastodonUI/View/Content/StatusAuthorView.swift:158]: Form a code-style perspective I would prefer, but that can be refactored later depending on what the team's stance on this is.

So far this LGTM!

```
actions += [
                .muteUser(.init(
                    name: menuContext.name,
                    isMuting: menuContext.isMuting
                )),
                .blockUser(.init(
                    name: menuContext.name,
                    isBlocking: menuContext.isBlocking
                )),
                .reportUser(
                    .init(name: menuContext.name)
                )
            ]
 ```

**Review by kimar — APPROVED** [review]
LGTM @nossipova 🎉 thank you very much for your contribution. This is going to be merged soon.

**Pull request merged**